calico-vpp-agent/cni/cni_server.go

```diff
@@ -378,9 +405,32 @@ func (s *Server) ServeCNI(t *tomb.Tomb) error {
 	}
 
 	pb.RegisterCniDataplaneServer(s.grpcServer, s)
+
+	netsSynced := make(chan bool)
+	nets := make(map[string]*watchers.NetworkDefinition)
+	go func() {
+		for t.Alive() {
+			event := <-s.cniMultinetEventChan
```

Comment:
We should probably also do a select
so that we don't wait on an empty chan on graceful teardown
````golang
select {
    case <-t.Dying():
        return
    case event := <-s.cniMultinetEventChan
        ...
} 
````